presto-hive/src/main/java/com/facebook/presto/hive/metastore/CachingHiveMetastore.java

```diff
@@ -261,23 +272,19 @@ public void flushCache()
     }
 
     @Override
-    public Optional<Map<String, HiveColumnStatistics>> getTableColumnStatistics(String databaseName, String tableName, Set<String> columnNames)
+    public Map<String, HiveColumnStatistics> getTableColumnStatistics(String databaseName, String tableName, Set<String> columnNames)
     {
-        Map<TableColumnStatisticsCacheKey, Optional<HiveColumnStatistics>> cacheValues =
+        Map<TableColumnStatisticsCacheKey, HiveColumnStatistics> cacheValues =
                 getAll(tableColumnStatisticsCache, columnNames.stream()
                         .map(columnName -> new TableColumnStatisticsCacheKey(databaseName, tableName, columnName))
-                        .collect(toList()));
-
-        return Optional.of(
-                ImmutableMap.copyOf(
-                        cacheValues.entrySet().stream()
-                                .filter(entry -> entry.getValue().isPresent())
-                                .collect(toMap(
-                                        entry -> entry.getKey().getColumnName(),
-                                        entry -> entry.getValue().get()))));
+                        .collect(toImmutableList()));
+
+        return cacheValues.entrySet().stream()
+                .filter(entry -> entry.getValue() != EMPTY_COLUMN_STATISTICS)
```

Comment:
HiveColumnStatistics cannot, normally, be compared by `==`, it's more a value object. Can we avoid comparing to some instance? What if we drop this filter altogether?

Reply:
EMPTY_COLUMN_STATISTICS is a marker instance, do cache empty value. Before that we used `Optional.empty()` for that purpose. `Remove column granularity from getStatistics* metastore methods` commit simplifies the caching a lot, and eliminates the need for this trick.

Reply:
I still do not like like. At least, annotate `EMPTY_COLUMN_STATISTICS` with
```
// Marker instance so that we can cache lack of stats.
```

Reply:
I remove this field in `Remove column granularity from getStatistics* metastore methods` commit. This is just a temporary code to make it compile.